src/directives/leaflet.js

```diff
@@ -24,9 +23,9 @@ angular.module("leaflet-directive", []).directive('leaflet',
         transclude: true,
         template: '<div class="angular-leaflet-map"><div ng-transclude></div></div>',
         controller: function ($scope) {
-            _leafletMap = $q.defer();
+            this._leafletMap = $q.defer();
```

Comment:
I would like to see where the cleanup code happens in angular, I want to verify it as it is good to know.